src/genie/libs/parser/iosxe/show_cable_tdr_interface.py

```diff
@@ -0,0 +1,154 @@
+'''show_cable_tdr_interface.py
+IOSXE parser for the following show command
+	* show cable tdr interface {interface}
+'''
+
+# Python
+import re
+import unittest
+
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Or, ListOf, \
+        Optional, And, Default, Use
+
+# import parser utils
+from genie.libs.parser.utils.common import Common
+
+# =================================================
+# Schema for:
+#   * 'show cable tdr interface {interface}'
+# ==================================================
+class ShowCableTdrIntSchema(MetaParser):
+    schema = {
+        Any(): {
+            Optional('interface'): str,
+            Optional('speed'): str,
+            Optional('date'): str,
+            Optional('time'): str,
+            Optional('pairs'): {
+                Any() : {
+                    Optional('length'): int,
+                    Optional('tolerance'): int,
+                    Optional('remote_pair'): str,
+                    Optional('status'): str,
+               },
+           },
+        }
+    }
+
+class ShowCableTdrInterface(ShowCableTdrIntSchema):
+    """
+    Parser for:
+            show cable tdr interface {interface}
+    """
+
+    cli_command = 'show cable tdr interface {interface}'
+
+    def cli(self, interface=None, output=None):
+        if output is None:
+            if interface:
+                output = self.device.execute(self.cli_command.format(interface=interface))
```

Comment:
Shouldn't interface be a mandatory field? What happens if interface is not provided?

```suggestion
    def cli(self, interface, output=None):
        if output is None:
            output = self.device.execute(self.cli_command.format(interface=interface))
```